test/utils/nodes/baseload.go

```diff
@@ -48,31 +49,13 @@ func GetLoad(k8sCli *kubernetes.Clientset, nodeName string) (Load, error) {
 
 	cpu := &resource.Quantity{}
 	mem := &resource.Quantity{}
-
 	for _, pod := range pods.Items {
-		// TODO: we assume a steady state - aka we ignore InitContainers
-		for _, cnt := range pod.Spec.Containers {
-			for resName, resQty := range cnt.Resources.Requests {
-				switch resName {
-				case corev1.ResourceCPU:
-					cpu.Add(resQty)
-				case corev1.ResourceMemory:
-					mem.Add(resQty)
-				}
-			}
-		}
+		req, _ := resourcehelper.PodRequestsAndLimits(&pod)
+		cpu.Add(req[corev1.ResourceCPU])
+		mem.Add(req[corev1.ResourceMemory])
 	}
-	// get full cpus, and always take even number of CPUs
-	cpuMillis, _ := cpu.AsInt64()
-	// we round the CPU consumption as expressed in millicores (not entire cores)
-	// in order to (try to) avoid bugs related to integer division
-	// int64(2900 / 1000) -> 2 -> roundUp(2, 2) -> 2 (correct, but unexpected!)
-	// OTOH
-	// roundUp(2900, 2000) -> 4000 -> 4000/1000 -> 4 (intended behaviour)
-	// TODO: we can use some testing of the test utilities here (!)
-	cpuAmount := roundUp(cpuMillis, 2000)
-	nl.CPU = *resource.NewQuantity(cpuAmount/1000, resource.DecimalSI)
 
+	nl.CPU = *resource.NewQuantity(cpu.Value(), resource.DecimalSI)
```

Comment:
why you simplified away the code?

Reply:
also it seems that quantity.Value() does the work for us of rounding up the cpus:
https://github.com/openshift-kni/numaresources-operator/blob/3289fb4fdb391b018fd98ec2fe2c5f95030d7c63/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go#L737

Reply:
uhm, something doesn't click here.
First of all, the error doesn't add up: in the code, we do
```
	nl.CPU = *resource.NewQuantity(cpuAmount/1000, resource.DecimalSI)
```
but cpuAmount is a int64, so from where this error is coming? `990m` is a string representation, which can't be hold in `cpuAmount`.
Second, the `Value()` function does indeed the round up, butin this code we do want to round up to cpu *pairs*, not to single integral CPU (this could have been more explicit in the comment, it is just hinted)

Reply:
> but cpuAmount is a int64, so from where this error is coming?
you are right, we don't have the error in the original version, so to make it clear why we got 0 CPU value after `AsInt64` I added a report based on the result of this function, which seems to return false as a second arg:
```
cpuMillis, ok := cpu.AsInt64()
	if !ok {
		return nl, fmt.Errorf("could not parse cpu=%s as int64", cpu)
	}
```
https://github.com/openshift-kni/numaresources-operator/blob/3289fb4fdb391b018fd98ec2fe2c5f95030d7c63/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go#L480



> Second, the Value() function does indeed the round up, butin this code we do want to round up to cpu pairs, not to single integral CPU (this could have been more explicit in the comment, it is just hinted)
do we want that in case we are dealing with SMT level 2? I may add the value to hit the upper even value of the cpus

Reply:
ok, so we need to be more careful in the conversion. Fixes to `baseload.go` need to be moved in their own commit (if not PR).
Rounding to HT pairs is meant to prevent any issue with HT alignment in cpumanager. Is not strictly needed, but should help. With modern servers with >= 32 cores per NUMA zone, should be no bother.

Reply:
all x86_64 cpus on the market or on the horizon all have even number of cores, so if we take care to reserve a even number, everything is nice and consistent. It's probably easier to just round up the baseload to multiples of 2